src/Routing/RouteSubscriber.php

```diff
@@ -16,22 +16,30 @@ class RouteSubscriber extends RouteSubscriberBase {
   /**
    * {@inheritdoc}
    */
-  protected function alterRoutes(RouteCollection $collection) {
-    // Remove these routes as to generate fatal errors wherever
-    // functionality is missing.
-    // @see user.routing.yml for original definition.
-    $routes_to_remove = [
+  protected function alterRoutes(RouteCollection $collection): void {
+
+    // Only the user 1 has access to user creation and deletion on Drupal.
+    $admin_routes = [
       'user.admin_create',
       'user.multiple_cancel_confirm',
+      'user.cancel_confirm',
+    ];
+    foreach ($admin_routes as $admin_route) {
+      if ($route = $collection->get($admin_route)) {
+        $route->setRequirement('_superuser_access_check', 'TRUE');
```

Comment:
For better readability, I think it's always better to move away the code from conditions.
It also reduces the cyclomatic complexity in overall.
So, in this case, I think it would be nicer to have:

```php
if (null === ($route = $collection->get($admin_route))) {
  continue;
}
$route->setRequirement('_superuser_access_check', 'TRUE'); 
```

As this is not a blocker nor defined in any PHPCS, it's up to the user to decide if he wants to do it or not, it's not mandatory.